### PR TITLE
:bug: Fish install broken

### LINKED ISSUE #234: Installer does not work on fish shell
**Describe the bug**
All I know for sure is that after following the install instructions, I cannot execute `okta-aws`. I've been debugging and I already notice one huge bug, which is that the installer added
```
export PATH="$HOME/bin:$PATH"
```
to my `config.fish`. That syntax is valid in `bash`, but in fish it should probably be
```
set -x PATH $HOME/bin $PATH
```
Furthermore, it didn't even create a `/bin` folder in my `$HOME` directory.

**To Reproduce**
1. Install the fish shell
2. Execute
```
curl 'https://raw.githubusercontent.com/oktadeveloper/okta-aws-cli-assume-role/master/bin/install.sh' | bash
```
3. Check your `config.fish`

### PR DESCRIPTION
**Pull request by AlainODea**
Problem Statement
-----------------
Issue #234 states:
> **Describe the bug**
> All I know for sure is that after following the install instructions, I cannot execute `okta-aws`. I've been debugging and I already notice one huge bug, which is that the installer added
> 
> ```
> export PATH="$HOME/bin:$PATH"
> ```
> 
> to my `config.fish`. That syntax is valid in `bash`, but in fish it should probably be
> 
> ```
> set -x PATH $HOME/bin $PATH
> ```
> 
> Furthermore, it didn't even create a `/bin` folder in my `$HOME` directory.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
>     1. Install the fish shell
> 
>     2. Execute
> 
> 
> ```
> curl 'https://raw.githubusercontent.com/oktadeveloper/okta-aws-cli-assume-role/master/bin/install.sh' | bash
> ```
> 
>     1. Check your `config.fish`

Solution
--------
 - Don't use export (it's a Bash thing, not fish)

 - Don't create withokta function (there's a withokta script)

Resolves #234

### CODE REVIEW

**Comment by AlainODea** [comment]
@patrickf3139 this should resolve your issue. It works for a fresh install using fish for me. Please test it and post a code review on this for me.

**Comment by patrickf3139** [comment]
Woohoo! It works! Thank you so much @AlainODea for the quick response and fix.

**Comment by patrickf3139** [comment]
My only suggestion, if you want to be really fish-, is to install `okta-aws` and `okta-sls` into `~/.config/fish/functions` as auto-loaded functions instead because
1. canonically, functions belong in the functions folder rather than config.fish
2. slight improvement to shell performance this way

But it's really not a big deal.

**Comment by AlainODea** [comment]
@patrickf3139 the fish functions directory is so much cleaner! Thank you for that feedback 👍

**Comment by AlainODea** [comment]
Please give this one last test to see that it works for you as well and then I'll merge it.

**Comment by patrickf3139** [comment]
Wait sorry, you create the function files without the .fish extension which is necessary for autoloading I think.

File: bin/install.sh [hunk 1]
```diff
@@ -47,27 +47,19 @@ function okta-sls {
...
 echo '
-#OktaAWSCLI
-export PATH="$HOME/bin:$PATH"
-function withokta
-    java -Djava.net.useSystemProxies com.okta.tools.WithOkta $argv
-end
 function okta-aws
     withokta "aws --profile $argv[1]" $argv
 end
+' > "${fishFunctionsDir}/okta-aws"
```
Comment (patrickf3139) [inline bin/install.sh]: append .fish

File: bin/install.sh [hunk 2]
```diff
@@ -47,27 +47,19 @@ function okta-sls {
...
+echo '
 function okta-sls
     withokta "sls --stage $argv[1]" $argv
 end
-' >> "${fishConfig}"
-fi
+' >> "${fishFunctionsDir}/okta-sls"
```
Comment (patrickf3139) [inline bin/install.sh]: append .fish

Comment (AlainODea) [reply bin/install.sh]: Right you are. I still had them in my config.fish.

**Comment by AlainODea** [comment]
Sorry about that, @patrickf3139. I've got it corrected now and re-tested with a clean config.fish.

**Comment by patrickf3139** [comment]
Tested, works. Thank you Alain. Nice working with you!

**Comment by AlainODea** [comment]
Nice working with you as well :) I hope this tool is useful for you (hopefully a bit friendlier to deploy now).

**Pull request merged**